components/uniset/src/uniset.rs

```diff
@@ -24,20 +25,20 @@ const BMP_MAX: u32 = 0xFFFF;
 /// and predefined ranges.
 #[derive(Debug, PartialEq, Hash, Eq, Clone, Yokeable, ZeroCopyFrom)]
 #[yoke(cloning_zcf)]
-pub struct UnicodeSet {
+pub struct UnicodeSet<'d> {
     // TODO: need advice - how should we remove Hash and Eq from UnicodeSet unless we need it?
 
     // If we wanted to use an array to keep the memory on the stack, there is an unsafe nightly feature
     // https://doc.rust-lang.org/nightly/core/array/trait.FixedSizeArray.html
     // Allows for traits of fixed size arrays
 
     // Implements an [inversion list.](https://en.wikipedia.org/wiki/Inversion_list)
-    inv_list: Vec<u32>,
+    inv_list: ZeroVec<'d, u32>,
     size: usize,
 }
 
 #[cfg(feature = "serde")]
-impl<'de> serde::Deserialize<'de> for UnicodeSet {
+impl<'de, 'd> serde::Deserialize<'de> for UnicodeSet<'d> {
```

Comment:
The Deserialize impl is the place where you need to perform the zero-copy magic.  You will need to update this impl in order to do that.